pkg/client/group_mgmt_client.go

```diff
@@ -411,9 +330,12 @@ func processResponse(client *GroupMgmtClient, response *resty.Response, path str
 				return nil, err
 			}
 		}
-		// process success response
-
-		return unwrapData(response.Body(), respHolder)
+		// unmarshall response data
+		dataIntf, totalRows, err := unwrapData(response.Body(), respHolder)
+		if params != nil && params.Page != nil && totalRows != nil {
```

Comment:
I guess the 'totalRows != nil' check doesn't really serve a purpose.

Reply:
It does since if there is error it will be nil .

Reply:
I'm saying this would be enough:

		if params != nil && params.Page != nil {
			params.Page.TotalRows = totalRows

If totalRows is nil, I would think params.Page.TotalRows should be nil also.

Reply:
I'm reconsidering this one.  I think what you mean is when an error occurs, we should preserve the prior TotalRows value even though we'll be returning something that didn't come back in the API response.  The client program will see that an error is returned and know not to inspect the TotalRows value.  That seems reasonable to me.  We can close this issue.